core/src/main/scala/com/danielwestheide/kontextfrei/rdd/RDDBase.scala

```diff
@@ -1,7 +1,25 @@
 package com.danielwestheide.kontextfrei.rdd
 
 import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
 
 private[kontextfrei] trait RDDBase {
   def sparkContext: SparkContext
+
+  def withSite[A, R](as: RDD[A])(body: RDD[A] => R): R = {
+
+    def isInKf(ste: StackTraceElement): Boolean = {
+      Option(ste.getClassName()).exists(_.startsWith("com.danielwestheide.kontextfrei"))
+    }
+
+    val (kf, user) = Thread.currentThread().getStackTrace()
+      .dropWhile(!isInKf(_))
+      .span(isInKf)
+
+    val method = kf.lastOption.map(_.getMethodName()).getOrElse("unknown")
+    val site = user.headOption.map(s => s"${s.getFileName()}:${s.getLineNumber()}").getOrElse("unknown")
+
```

Comment:
Maybe all the logic in this method could be extracted into a pure function returning a `String`, to make it easier to test. All this method would do then is to call that function, set its result as the call site and invoke the body. What do you think?

Reply:
Yes, will do, I should have thought of that, Sorry but I'm new to this - should I amend the previous commit or make a second commit?

Reply:
No need to apologize at all. :) I haven't come up with any rules for this yet, and everyone seems to have their own preferences. For me, it's fine if you just add another commit.